core/src/main/java/org/apache/calcite/rel/RelNode.java

```diff
@@ -246,6 +246,8 @@ default String getDigest() {
     return getRelDigest().toString();
   }
 
+  String getTreeDigest();
```

Comment:
This is what I meant that is not very neat; adding a new API in `RelNode` just for debugging purposes. 
Nevertheless if we decide to go this way I think it would be better to use the following signature:
`String explain();`

Going one step further we could make it a default method with an implementation similar to the one below:
```
default String explain() {
    StringWriter sw = new StringWriter();
    RelWriter planWriter =
        new RelWriterImpl(
            new PrintWriter(sw), SqlExplainLevel.EXPPLAN_ATTRIBUTES, false);
    explain(planWriter);
    return sw.toString();
}
```
Even more we could deprecate/remove `RelOptUtil#toString(RelNode)` variants in favor of this one.

Reply:
@zabetak +1
I also think it makes sense to deprecate/remove `RelOptUtil#toString(RelNode)` since there is also a similar (more "powerful") method that covers that functionality: `RelOptUtil#dumpPlan`.
We could even consider to use it in the default implementation of the new method for the sake of code re-usability:
````
default String explain() {
    return RelOptUtil.dumpPlan("", this, SqlExplainFormat.TEXT, SqlExplainLevel.EXPPLAN_ATTRIBUTES);
}
````

Reply:
Hi fellows
I've updated my code, would you please take a look?
@zabetak Julian commented in the JIRA: "The implementation should continue to live in RelOptUtil or wherever", so I just still call RelOptUtil.toString
@rubenada "RelOptUtil.dumpPlan("", this, SqlExplainFormat.TEXT, SqlExplainLevel.EXPPLAN_ATTRIBUTES)"'s output seems the same with RelOptUtil.toString.